enhancements/sha-bundle-digest.md

```diff
@@ -0,0 +1,136 @@
+---
+title: SHA-Bundle-Digest
+authors:
+  - '@theishshah'
+reviewers:
+  - '@camilamacedo86'
+  - '@joelanford'
+  - '@jmrodri'
+  - '@ryantking'
+  - '@tlwu2013'
+approvers:
+  - '@tlwu2013'
+  - '@jmrodri'
+  - '@joelanford'
+
+creation-date: 2022-01-14
+
+last-updated: 2022-01-14
+
+status: implementable
+---
+
+# Support SHA-based Image Tags
+
+## Summary
+
+Currently, operators generated by `operator-sdk` will store and retrieve container images based on the image name and
+the image tag in the bundle. This EP adds support to `operator-sdk` such that images can be stored and retrieved in the
+bundle with an image name and an image digest as an optional replacement for a tag. Image digests are SHA-256 sums that
+are hashed from the image layer therefore uniquely identifying one specific image.
+
+## Motivation
+
+Many users run Kubernetes in an air-gapped environment (also known as an offline or disconnected environment), which is
+when a cluster network does not have any access to the public internet, usually for security purposes. In this setup,
+the digest of an image will be different than it is in the public repository since the image location changes. For
+example, if a user imports the image `quay.io/golang:1.17` into their air-gapped registry as
+`quay.internaldomain.net/golang:1.17`, the digest of the image will be different due to the different name even though
+the contents of the images are identical. Using digest SHAs instead of tags in the air-gapped environment will keep the
+digest from changing registry to registry.
```

Comment:
```suggestion
Many users run Kubernetes in an air-gapped environment (also known as an offline or disconnected environment), which is
when a cluster network does not have any access to the public internet, usually for security purposes. In this setup,
since the image is hosted in a registry within the air-gapped environment, it will have a different name. For example,
if you push `golang:1.17.6` to an internal registry, it will have a name along the lines of
`registry.internaldomain.net/library/golang:1.17.6`. Since the name, which is part of the image manifest, is different,
the image digest will be different. This is a problem for security-conscience users who want to ensure that they are
using the exact image they expect since by definition the digests of the same image in two different registries will be
different. This EP resolves that by supporting an operator configuration that works directly with image digests instead
of tags. Users will then be able to push images to their internal registry using the image's digest instead of its tag,
which will keep the digest from changing, and then the operator will pull the image based on its digest.
```